File: test/corpus/ruby/clean.rb
# Cohesive: each method calls the next, so the file is one connected component.

LO = 0.0
HI = 1.0

def clampv(x, lo, hi)
  if x < lo
    return lo
  end
  if x > hi
    return hi
  end
  x
end

def normalize(x)
  clampv(x, LO, HI)
end

def scale(x, k)
  normalize(x) * k
end

def accumulate(xs)
  total = 0.0
  xs.each do |x|
    total += scale(x, 2)
  end
  total
end

File: test/corpus/c/clean.c
// Cohesive: each function calls the next, so the file is one connected component.

static const double LO = 0.0;
static const double HI = 1.0;

double clampv(double x, double lo, double hi) {
    if (x < lo) {
        return lo;
    }
    if (x > hi) {
        return hi;
    }
    return x;
}

double normalize(double x) {
    return clampv(x, LO, HI);
}

double scale(double x, double k) {
    return normalize(x) * k;
}

double accumulate(const double *xs, int n) {
    double total = 0.0;
    for (int i = 0; i < n; i++) {
        total += scale(xs[i], 2);
    }
    return total;
}
